### DISCUSSION TITLE
What about NextJS 13 app dir ?

### DISCUSSION THREAD

**Discussion by FrenchIsNow** [tolgee/tolgee-platform#1776] [Q&A]
Hi , I wanted to know if Tolgee is compatible with the last NextJs App version ? If yes how can I implement it in my projet ? 

Thanks for your time.

**Comment by stepan662** [comment]
Hey, not fully. But I think it's possible to use `TolgeeProvider` in root layout, so tolgee can work as before.
I'll have to check how to use it with the server components, but it should be possible with the plain `tolgee.t` function.
However the in-context editing will be a bit problematic with the server components.

**Reply by ferhatatagun** [reply]
Hi, When will you provide this?  On the server side, router-based problems trigger many problems and make them unusable.

Thanks so much for your time.
